src/util/http/UrlParser.h

```diff
@@ -14,51 +15,30 @@
 
 namespace ad_utility {
 /**
- * /brief A simple parser for urls/uris. It does NOT provide the full
- * URI specification, but only a small subset of it.
- *
- * TODO<joka921> Replace by a suitable third-party library when we have
- * chosen one. My current favourite is Boost::Url by Vinnie Falco, but there
- * exists no stable release yet.
+ * /brief Some helpers to parse request URLs in QLever.
  */
 class UrlParser {
  public:
-  /// Representation of the "path" and "query" of a URL. For a GET request, the
-  /// "path" is the part before the "?" (or everything if there is no "?"), and
-  /// the "query" is the part after the "?" (empty if there is no "?"). The
-  /// key-value pairs of the "query" are stored in a hash map.
-  struct UrlPathAndParameters {
-    std::string _path;
-    ad_utility::HashMap<std::string, std::string> _parameters;
+  // Representation of parsed request URL.
+  // - `path_` is the URL path.
+  // - `query_` and `update_` contain the Query or Update respectively. Only one
+  // of the two will be non-empty.
+  // - `parameters_` is a hashmap of the URL query parameters.
+  struct ParsedUrl {
+    std::string path_;
+    ad_utility::HashMap<std::string, std::string> parameters_;
+    std::optional<std::string> query_;
+    std::optional<std::string> update_;
   };
```

Comment:
One probably wants another type that only contains `path_` and `parameters_` for `parseRequestTarget`.

Reply:
Decided against that. The overhead of moving the fields around seems not worth it.